### PR TITLE
fix validation bug

### PR DESCRIPTION
**Pull request by danhuynhdev**
The bug occurs in later chapters where length validation is ignore
when update attributes. This is because the previous commit make 
validation only run on create. This commit fix the bug by split the validation
to 2 part, remove the `on: :create` option, and instead only check presence 
of password when the password is changed or create new User.

### CODE REVIEW

**Comment by danhuynhdev** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -8,7 +8,9 @@ class User < ApplicationRecord
             uniqueness: {case_sensitive: false}
   validates :name, length: {maximum: Settings.max_name_length}, presence: true
   validates :password, length: {minimum: Settings.min_password_length},
-            on: :create, presence: true
+            if: ->{password.present?}
+  # only check presence of password when password is changed or create new User
+  validates :password, presence: true, if: :new_password?
```
Comment (dothidiemthao) [inline app/models/user.rb]: cái này nên đưa lên trước validates length nhé

**Comment by dothidiemthao** [comment]
commented

**Comment by danhuynhdev** [comment]
ready

**Pull request merged**